tests/datatypes/test_str.py

```diff
@@ -30,6 +30,13 @@ def test_isspace(self):
                 print(s.isspace())
             """)
 
+    def test_isalpha(self):
+        self.assertCodeExecution("""
+            for s in ['Hello World', 'hello wORLd.', 'Hello world.', '',
```

Comment:
can you add a test for `hello1` ?
I think this code also has a off-by-one issue.